src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -140,12 +141,23 @@ const Body = BaseForm.extend(Object.assign(
       };
 
       const doProbing = () => {
-        return checkPort()
+        this.checkPortXhr = checkPort();
+        return this.checkPortXhr
           // TODO: can we use standard ES6 promise methods, then/catch?
-          .done(onPortFound)
+          .done(() => {
```

Comment:
I think you can improve this using `ES6 promise methods` now ?

Reply:
I think we should do it later. It requires some more investigation and testing on how to convert the jquery ajax with timeout option to use promise so we can use then/catch.

Reply:
Sorry, I don't get it. What's so different about this ?

Reply:
This is out of scope for the fix considering the time urgency. This should be tackled along with https://oktainc.atlassian.net/browse/OKTA-276652.